File: 30-days-of-code/day_25.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

bool isPrime(long N) {
    if (N == 1) return false; 
    if (N == 2) return true; 
    if ((N & 0x1) == 0x0) return false; 
    
    long rootN = (long) sqrt(N);
    
    for (long divisor = 3; divisor <= rootN; divisor += 2) {
        if ((N % divisor) == 0) return false;
    }
    
    return true;
}


int main() {
    int n, num;
    cin >> n; 
    for(int i=0; i<n; i++) {
        cin >> num;
        if (isPrime(num) == true){
            cout << "Prime" << endl;
        } else {
            cout << "Not prime" << endl;
        }
    }
    return 0;
}

File: 30-days-of-code/day_25.py
import math

def isPrime(n):
    if (n==1): 
        return False
    if (n==2): 
        return True
    if ((n & 0x1) == 0x0):
        return False

    root = math.sqrt(n)
    
    for i in range(3, int(root+1), 2):
        if n%i == 0:
            return False

    return True

n = int(input())
for i in range(n):
    num = int(input())
    if isPrime(num) == True:
        print('Prime')
    else:
        print("Not prime")
